sqlglot/dialects/clickhouse.py

```diff
@@ -593,6 +598,28 @@ def _parse_partition(self) -> t.Optional[exp.Partition]:
 
             return self.expression(exp.Partition, expressions=expressions)
 
+        def _parse_alter_table_replace(self) -> t.Optional[exp.Expression]:
+            partition = self._parse_partition()
+
+            if not partition or not self._match(TokenType.FROM):
+                return None
+
+            db = None
+            table = self._parse_id_var(any_token=False)
+
+            if self._match(TokenType.DOT):
+                db = table
+                table = self._parse_id_var(any_token=False)
+
+            if not table:
+                return None
+
+            return self.expression(
+                exp.ReplacePartition,
+                expression=partition,
+                source=self.expression(exp.Table, this=table, db=db),
+            )
```

Comment:
```suggestion
            return self.expression(
                exp.ReplacePartition, expression=partition, source=self._parse_table_parts()
            )
```

Reply:
I wanted to use this one, but ClickHouse allows only identifiers here, and `self._parse_table_parts()` allows to use string literals as well. If you think that would be okay, then I'll fix it if course

Reply:
Yep it's ok, sqlglot's parser is generally lenient because it doesn't aim to validate syntax

Reply:
OK then! Will keep that in mind as well :)